apkid/rules/dex/obfuscators.yara

```diff
@@ -187,3 +187,21 @@ rule allatori_demo : obfuscator
   condition:
     $s and is_dex
 }
+
+rule aamo : obfuscator
+{
+	meta:
+		description = "aamo obfuscator"
```

Comment:
and...what about `AAMO (string encryption)`? Are you planning to fingerprint more features from this obfuscator?

Reply:
Yes, i'll try to fingerprint more features;  but I was wondering whether it makes more sense or not to have all combined in one single rule? or is it better to have 1 feature == 1 rule?

instead of having a lot of ```AAMO (feature1)```, ```AAMO (feature2)```, ```AAMO (feature3)``` and so no

Reply:
What is better: 1 feature == 1 rule? which means that we will have a lot of ```AAMO (feature-N```. Isn't it better to update this rule to solely detect AAMO?

Reply:
I think there's a range of acceptable levels of detail -- it depends on the cost / benefit. If fingerprinting individual features is easy (meaning the rule has a low running cost and is unlikely to ever FP), and it's a common obfuscator (high benefit), then that's good. It becomes less beneficial to fingerprint every possible permutation configuration for uncommon obfuscators. No idea how common this one is in the wild.

In general, the *minimum* level of detail is to fingerprint the obfuscator regardless of features. And if at all possible, make distinctions between major versions. Of course, it's not always possible to know the actual product version number, and in those cases the rule is to use typical malware variant naming conventions, i.e. use the alphabet, e.g. "a", "b", "c" .

Is string encryption optional for this obfuscator? Or is it always turned on?

Reply:
It is optional.